### PR TITLE
Enhancement(popover): introduced new props to facilitate delay in open/close transition in popover component 

### PR DESCRIPTION
**Pull request by mariappan-subramanian**
There was a need to control the popover open/close transition for Marketplace apps website. Since there was no timing control over these transition it created kind of glitch in the interaction & faced usability issues. Thus introduced additional props to control these transition as in element-plus popover component (https://element.eleme.io/#/en-US/component/popover) which we used earlier in the project.

fix: #666

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
Local

### CODE REVIEW

**Comment by mariappan-subramanian** [comment]
Closed the [previous PR](https://github.com/freshworks/crayons/pull/725) as it got messed up a bit with many force pushes, thus created a fresh PR with the same set of reviewed changes @kulkarni-kashyap 

Please review.

File: packages/crayons-core/src/index.html [hunk 1]
```diff
@@ -13,6 +13,25 @@
   </head>
   <body>
     <fw-button expand color="primary">Primary Button</fw-button>
+    <!-- <fw-popover trigger="hover" hide-after="2000" show-after="2000">
```
Comment (kulkarni-kashyap) [inline packages/crayons-core/src/index.html:16]: Can you please revert the changes in index.html ? We are good to go otherwise!

**Review by kulkarni-kashyap — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/index.html (hunk 1 above)
Comment (mariappan-subramanian) [reply packages/crayons-core/src/index.html:16]: Sure, will do

File: packages/crayons-core/src/components/popover/popover.tsx [hunk 2]
```diff
@@ -110,64 +119,30 @@ export class Popover {
...
-      if (this.trigger !== 'hover') {
-        this.overlay.style.display = 'block';
-      }
-      this.isOpen = !this.isOpen;
-      if (this.contentRef?.tagName === 'FW-LIST-OPTIONS') {
-        const listOptionsElement = this.contentRef as HTMLFwListOptionsElement;
-        listOptionsElement.scrollToLastSelected();
+      clearTimeout(this.timerId);
+      if (this.showAfter > 0) {
+        this.timerId = setTimeout(
```
Comment (arvindanta) [inline packages/crayons-core/src/components/popover/popover.tsx:124]: Can you create a function called delay 

```
async delay(ms) {  return new Promise((res) => { this.timerId = setTimeout(res,ms))  }}
```
and use this function in `show` method. 

```
if(!this.isOpen) {
clearTimeout(this.timerId);

if(this.showAfter> 0) await delay(this.showAfter)
```
.....
you can keep the rest of the code here instead of moving it to an another function

}

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/components/popover/popover.tsx (hunk 2 above)
Comment (mariappan-subramanian) [reply packages/crayons-core/src/components/popover/popover.tsx]: Done

**Comment by mariappan-subramanian** [comment]
@arvindanta @kulkarni-kashyap Please re-review this PR. I've made the suggested changes but not able to re-request review from both at the same time (not sure why, it keeps swapping). Thus making a note here.

**Review by arvindanta — APPROVED** [review]

**Review by arvindanta — APPROVED** [review]

**Review by arvindanta — CHANGES_REQUESTED** [review]

**Review by arvindanta — APPROVED** [review]

**Comment by arvindanta** [comment]
LGTM
